### PR TITLE
Remove rts_hooks.c in favor of --with-rtsopts

### PR DESCRIPTION
**Pull request by bpfoley**
`rts_hooks` is only used to set the default RTS for bsc, bluetcl and
the extra tools, and all of this can be done more easily with
--with-rtsopts, instead of having extra Makefile machinery.

--with-rtsops has been present since at least GHC 7.0.1.

### CODE REVIEW

File: src/comp/Makefile [hunk 1]
```diff
@@ -172,7 +172,9 @@ GHCINCLUDES =  \
 	-i$(HTCL_HS)
 GHCTMP = '-tmpdir $(TMPDIR)'
 # Default RTS flags for programs built with a Haskell main
-RTSFLAGS =
+# 25GMB heap, 10MB stack, 1 second interval between heap profiles
```
Comment (quark17) [inline src/comp/Makefile:175]: Should that 'G' be '6'?

File: src/comp/Makefile [hunk 2]
```diff
@@ -290,78 +287,79 @@ BLUETCL_BUILDLIBS = \
...
 
 .PHONY: showrules
-showrules: $(SOURCES) $(EXTRAOBJS)
+showrules: $(SOURCES)
 	$(PREBUILDCOMMAND)
-	$(BUILDCOMMAND) -main-is Main_$@ $(BUILDFLAGS) $(RTSFLAGS) $(EXTRAOBJ)
+	$(BUILDCOMMAND) -main-is Main_$@ $(BUILDFLAGS) $(RTSFLAGS)
 	$(POSTBUILDCOMMAND)
 
+
```
Comment (quark17) [inline src/comp/Makefile:351]: Is this extra blank line intended?

File: src/comp/Makefile [hunk 3]
```diff
@@ -290,78 +287,79 @@ BLUETCL_BUILDLIBS = \
...
+# bluetcl is built in two steps:
+# 1. Build BlueTcl module using -c to emit .hi/.o files, and also the
+# BlueTcl_stub.h reqired by bluetcl_Main.hsc in the next action
```
Comment (quark17) [inline src/comp/Makefile:354]: Can you indent the second line three spaces?

File: src/comp/Makefile (hunk 1 above)
Comment (bpfoley) [reply src/comp/Makefile]: Woops.

File: src/comp/Makefile (hunk 2 above)
Comment (bpfoley) [reply src/comp/Makefile]: Nope. My bad.

File: src/comp/Makefile (hunk 3 above)
Comment (bpfoley) [reply src/comp/Makefile]: Done.

**Pull request merged**